logger/logger_writer.go

```diff
@@ -160,41 +187,33 @@ func queueMsg(lvl Level, prefix, format string, v ...interface{}) (err error) {
 
 	msg.time = time.Now()
 
+	onErr := func() {
+		atomic.AddUint64(&errCount, 1)
+		_ = freeMsg(msg) // ignore error
```

Comment:
GoLand suggested assigning the return value to `_`. I added the comment.

If explicitly ignoring the error will lead to noisy code does that imply
* we're not checking errors enough
* we're returning too many errors
* both a & b
* none of above

Regardless, I agree that `freeMsg` should never return an error. If it did, something is **very** corrupt in memory, in which case, shouldn't we abort/quit and let the OS's services layer restart it?

If that's too risky, I'm OK w/ removing the comment. But can we leave the assignment to `_` (I've a bit of OCD with IDE warnings)? It is just one line in this  code. 🙂

Reply:
I don't have a super-strong feeling about this.  But in just this file, I also see ignored error returns from 
https://github.com/kentik/golog/blob/6cd628aab0dcaa548ed492454c19626a44e7fb04/logger/logger_writer.go#L256
and from 
https://github.com/kentik/golog/blob/6cd628aab0dcaa548ed492454c19626a44e7fb04/logger/logger_writer.go#L293

Are these exempted from the IDE warning?  I'm not convinced that we improve the codebase much by either explicitly ignoring errors everywhere we call net.Conn.Close()/os.File.Close(), or by handling them (which probably just means logging them, and then training ops to ignore the logs...)

Am I being too cynical?

Reply:
Including the two you mentioned, the IDE shows total of 6 "Unhandled error" warnings on this file 🙂:
![image](https://user-images.githubusercontent.com/963723/159816227-948adb52-6797-48ca-9aae-11b7bf682725.png)

I left them as-is since I didn't modify those parts of the code.